### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by raphlinus**
Update various dependencies to latest.

### CODE REVIEW

File: piet-gpu-hal/Cargo.toml [hunk 1]
```diff
@@ -7,12 +7,12 @@ license = "MIT/Apache-2.0"
...
 [dependencies]
-ash = "0.33"
-ash-window = "0.7"
-raw-window-handle = "0.3"
-bitflags = "1.2.1"
-smallvec = "1.6.1"
-bytemuck = "1.7.2"
+ash = "0.37"
+ash-window = "0.11"
+raw-window-handle = "0.5"
```
Comment (m-hugo) [inline piet-gpu-hal/Cargo.toml:12]:
```suggestion
raw-window-handle = "0.4"
```
needs to match the dependencies

File: piet-gpu-hal/Cargo.toml [hunk 2]
```diff
@@ -7,12 +7,12 @@ license = "MIT/Apache-2.0"
 edition = "2018"
 
 [dependencies]
-ash = "0.33"
-ash-window = "0.7"
-raw-window-handle = "0.3"
-bitflags = "1.2.1"
-smallvec = "1.6.1"
-bytemuck = "1.7.2"
+ash = "0.37"
```
Comment (m-hugo) [inline piet-gpu-hal/Cargo.toml:10]:
```suggestion
ash = {version = "0.37", features=["linked"]}
```
current behavior is now opt-in

File: piet-gpu/Cargo.toml [hunk 3]
```diff
@@ -28,11 +28,11 @@ path = "../piet-gpu-types"
...
 [dependencies]
 piet = "0.2.0"
-png = "0.16.2"
-rand = "0.7.3"
+png = "0.17.6"
+rand = "0.8.5"
 roxmltree = "0.13"
-winit = "0.26.1"
-clap = "2.33"
+winit = "0.27.3"
```
Comment (m-hugo) [inline piet-gpu/Cargo.toml:34]:
```suggestion
winit = {version = "0.27.3",  default-features = false, features = ["x11", "wayland", "wayland-dlopen"]}
```
remove added bloat

**Review by m-hugo — CHANGES_REQUESTED** [review]
also in vulkan.rs:

-let mut exts = Extensions::new(entry.enumerate_instance_extension_properties()?);
+let mut exts = Extensions::new(entry.enumerate_instance_extension_properties(None)?);

-let entry = Entry::new()?;
+let entry = Entry::linked();

-.message_type(vk::DebugUtilsMessageTypeFlagsEXT::all())
+.message_type(vk::DebugUtilsMessageTypeFlagsEXT::from_raw(0b111))

File: piet-gpu-hal/Cargo.toml [hunk 4]
```diff
@@ -7,12 +7,12 @@ license = "MIT/Apache-2.0"
...
+ash-window = "0.11"
+raw-window-handle = "0.5"
+bitflags = "1.3.2"
```
Comment (bzm3r) [inline piet-gpu-hal/Cargo.toml:13]: Do we need to specify the minor version here? See also for other crates in this and piet-gpu's Cargo.toml

File: piet-gpu-hal/Cargo.toml (hunk 1 above)
Comment (bzm3r) [reply piet-gpu-hal/Cargo.toml:12]: @m-hugo Which dependencies are using `0.4`? `ash-window`?

File: piet-gpu-hal/Cargo.toml (hunk 2 above)
Comment (raphlinus) [reply piet-gpu-hal/Cargo.toml:10]: I don't think we depend on the "linked" behavior. Perhaps you can explain why you think we do?

File: piet-gpu-hal/Cargo.toml [hunk 5]
```diff
@@ -7,12 +7,12 @@ license = "MIT/Apache-2.0"
...
 
 [dependencies]
-ash = "0.33"
-ash-window = "0.7"
-raw-window-handle = "0.3"
-bitflags = "1.2.1"
-smallvec = "1.6.1"
-bytemuck = "1.7.2"
+ash = "0.37"
+ash-window = "0.11"
```
[the remark this answers is not in the document]
Comment (raphlinus) [reply piet-gpu-hal/Cargo.toml:11]: It was released 2 days after I first made this commit. I'll try upgrading to it, or we can just commit this and refine later. I'm inclined to wait until the points raised by @m-hugo are settled, though I don't see any that I consider truly blocking merging this.

File: piet-gpu-hal/Cargo.toml (hunk 1 above)
Comment (m-hugo) [reply piet-gpu-hal/Cargo.toml:12]: > @m-hugo Which dependencies are using `0.4`? ash-window 0.11?

yes

File: piet-gpu-hal/Cargo.toml (hunk 2 above)
Comment (m-hugo) [reply piet-gpu-hal/Cargo.toml:10]: > I don't think we depend on the "linked" behavior. Perhaps you can explain why you think we do?

I don't know your internals but 
Entry::new()?
was renamed
Entry::linked()
and hidden behind this flag

File: piet-gpu-hal/Cargo.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (m-hugo) [reply piet-gpu-hal/Cargo.toml:11]: > It was released 2 days after I first made this commit. I'll try upgrading to it, or we can just commit this and refine later. I'm inclined to wait until the points raised by @m-hugo are settled, though I don't see any that I consider truly blocking merging this.
well it doesn't compile unless you upgrade ash-window or downgrade  raw-window-handle

[the remark this answers is not in the document]
Comment (m-hugo) [reply piet-gpu-hal/Cargo.toml:11]: by it I mean cargo run --bin winit
(of course android.rs is now even more broken but I'm not sure what it's used for, winit.rs works fine on android)

File: piet-gpu-hal/Cargo.toml (hunk 4 above)
Comment (raphlinus) [reply piet-gpu-hal/Cargo.toml:13]: We don't need to, but I generally do.

File: piet-gpu-hal/Cargo.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (raphlinus) [reply piet-gpu-hal/Cargo.toml:11]: For some reason, the PR as uploaded is working on my mac, but I am seeing failures on Windows, so I will be sure to test there. I'll look into this and upload another commit, hopefully later today.

I'd be happy to remove android.rs if vanilla winit works. Can you share the steps you used to compile it and if those depend on any installed tools (cargo apk) and if so what version? This stuff has been pretty fragile for me.

[the remark this answers is not in the document]
Comment (m-hugo) [reply piet-gpu-hal/Cargo.toml:11]: ash-specific code must be hidden behind a platform flag then

I can send a pr  later for android support, it's just cargo apk + the right ndk_glue (0.7 for 0.27) + some main() entrypoint hacks

**Comment by raphlinus** [comment]
Ok, I've made some mostly ash-related changes and it now works on Vulkan + Windows. I haven't tested yet on Linux but don't expect serious problems. I also haven't updated the android example; I'll probably do that so that the PR is clean, but as I mentioned I am interested in combining that into the winit binary.

**Review by m-hugo — APPROVED** [review]
- winit.rs on linux
- android.rs on android

build and run properly

but without my winit features change, it builds slower than master

**Comment by m-hugo** [comment]
I've got an achievement for having co-authored a one-line commit, lmao

and apparently even Contributor is not enough to allow me to resolve my own reviews, I need to be Collaborator with write-access, GitHub's sane defaults…

**Pull request merged**